src/videojs-hls.js

```diff
@@ -392,6 +392,37 @@ videojs.Hls.prototype.addCuesForMetadata_ = function(segmentInfo) {
   minPts = Math.min(isFinite(segment.minVideoPts) ? segment.minVideoPts : Infinity,
                     isFinite(segment.minAudioPts) ? segment.minAudioPts : Infinity);
 
+  Cue = window.WebKitDataCue || window.VTTCue;
+
+  deprecateOldCue = function(cue) {
+    if (cue.frame.id) {
+      Object.defineProperty(cue.frame, 'id', {
```

Comment:
Also, we could use `Object.defineProperties` here, though that shouldn't hold up this PR:

```js
Object.defineProperties(cue.frame, {
  'id': {
    get: () => cue.value.key
  },
  value: {
    get: () => cue.value.data
  },
  privateData: {
    get: () => cue.value.data
  }
});
```